<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>VoronoiDialog</class>
 <widget class="QDialog" name="VoronoiDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>411</width>
    <height>500</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <property name="sizeConstraint">
    <enum>QLayout::SetFixedSize</enum>
   </property>
   <item row="6" column="0">
    <widget class="QRadioButton" name="radio_random">
     <property name="text">
      <string>random</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>type</string>
     </property>
    </widget>
   </item>
   <item row="4" column="2">
    <widget class="QSpinBox" name="number_of_points">
     <property name="maximum">
      <number>65536</number>
     </property>
    </widget>
   </item>
   <item row="8" column="0">
    <widget class="QRadioButton" name="radio_square">
     <property name="text">
      <string>square</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>radius (in %)</string>
     </property>
    </widget>
   </item>
   <item row="8" column="1">
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>width (in %)</string>
     </property>
    </widget>
   </item>
   <item row="9" column="1">
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>length (in %)</string>
     </property>
    </widget>
   </item>
   <item row="10" column="1">
    <widget class="QLabel" name="label_9">
     <property name="text">
      <string>length (in %)</string>
     </property>
    </widget>
   </item>
   <item row="16" column="1">
    <widget class="QDoubleSpinBox" name="initial_y">
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
    </widget>
   </item>
   <item row="16" column="0">
    <widget class="QLabel" name="label_11">
     <property name="text">
      <string>initial y</string>
     </property>
    </widget>
   </item>
   <item row="17" column="0">
    <widget class="QLabel" name="label_12">
     <property name="text">
      <string>initial angle</string>
     </property>
    </widget>
   </item>
   <item row="15" column="0">
    <widget class="QLabel" name="label_10">
     <property name="text">
      <string>initial x</string>
     </property>
    </widget>
   </item>
   <item row="17" column="1">
    <widget class="QSpinBox" name="initial_angle"/>
   </item>
   <item row="1" column="2">
    <widget class="QComboBox" name="voronoi_type">
     <item>
      <property name="text">
       <string>nearest distance</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>two nearest ratio</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="20" column="2">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>point place</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>metric</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>number of points</string>
     </property>
    </widget>
   </item>
   <item row="13" column="0">
    <widget class="QLabel" name="label_4">
     <property name="maximumSize">
      <size>
       <width>100</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="text">
      <string>connect L-system to input to use it for point placement</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="11" column="0">
    <widget class="QRadioButton" name="radio_custom">
     <property name="text">
      <string>custom</string>
     </property>
    </widget>
   </item>
   <item row="15" column="1">
    <widget class="QDoubleSpinBox" name="initial_x">
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
    </widget>
   </item>
   <item row="11" column="2">
    <widget class="QPushButton" name="clear_button">
     <property name="text">
      <string>clear</string>
     </property>
    </widget>
   </item>
   <item row="13" column="2">
    <widget class="PointEditFrame" name="point_editor">
     <property name="minimumSize">
      <size>
       <width>200</width>
       <height>200</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>200</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
    </widget>
   </item>
   <item row="9" column="0">
    <widget class="QRadioButton" name="radio_cross_horizontal">
     <property name="text">
      <string>cross horizontal</string>
     </property>
    </widget>
   </item>
   <item row="10" column="0">
    <widget class="QRadioButton" name="radio_cross_diagonal">
     <property name="text">
      <string>cross diagonal</string>
     </property>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="QComboBox" name="voronoi_metric">
     <item>
      <property name="text">
       <string>Euclidean</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>taxicab</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Chebyshev</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QRadioButton" name="radio_circle">
     <property name="text">
      <string>circle</string>
     </property>
    </widget>
   </item>
   <item row="14" column="0">
    <widget class="QLabel" name="label_13">
     <property name="text">
      <string>L-system</string>
     </property>
    </widget>
   </item>
   <item row="7" column="2">
    <widget class="QSpinBox" name="radius">
     <property name="maximum">
      <number>100</number>
     </property>
    </widget>
   </item>
   <item row="8" column="2">
    <widget class="QSpinBox" name="width">
     <property name="maximum">
      <number>100</number>
     </property>
    </widget>
   </item>
   <item row="9" column="2">
    <widget class="QSpinBox" name="length1">
     <property name="maximum">
      <number>100</number>
     </property>
    </widget>
   </item>
   <item row="10" column="2">
    <widget class="QSpinBox" name="length2">
     <property name="maximum">
      <number>100</number>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PointEditFrame</class>
   <extends>QFrame</extends>
   <header>pointeditframe.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>VoronoiDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>VoronoiDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
